infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironmentFactory.java

```diff
@@ -103,6 +107,7 @@ protected KubernetesEnvironment doCreate(
         Pod pod = (Pod) object;
         pods.put(pod.getMetadata().getName(), pod);
```

Comment:
The NPE you refer to in the TODO below can also happen here.

Reply:
it can not. See https://github.com/eclipse/che/blob/448c6f3b0fd2cae70b300402b6701d259788bf44/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironmentFactory.java#L102-L104

Reply:
but pods are not required to have a name by the Kubernetes API. They need to either have a name or generateName, I believe.

Reply:
Feel free to create an issue to improve k8s/os infrastructure to respect `generateName` if `Kubernetes API` really does not require a name. As far as I see OpenShift requires `name` even if `generateName` is specified 
![Screenshot_20190318_091601](https://user-images.githubusercontent.com/5887312/54513008-af669580-495e-11e9-8f42-a9fbef1c6cf5.png)